docs/tool/js/views/filter-view.js

```diff
@@ -32,8 +32,36 @@ var filterView = {
         var allPills = oldPills.enter().append("div")
             .attr("class","ui label transition visible")
             .classed("clear-single",true)
-          .merge(oldPills)
-            .text(function(d) { return d['component']['display_name'];});
+            // Animate a label that 'flies' from the filter component
+            // to the pillbox.
+            .text(function(d) { return d['component']['display_name'];})
+            .each(function(d){
+                var originElement = document.getElementById("filter-content-"+d.component.source);
+                var destinationElement = this;
+                var flyLabel = document.createElement('div');
+                flyLabel.textContent = this.textContent;
+                console.log("flyer this", this);
+                flyLabel.classList.add('ui', 'label', 'transition', 'visible', 'clear-single-flier');
+                var originRect = originElement.getBoundingClientRect();
+                flyLabel.style.left = originRect.left + 'px';
+                flyLabel.style.top = ((originRect.top + originRect.bottom)/2) + 'px';
+                document.body.appendChild(flyLabel);
+                console.log("flyLabel after append", flyLabel);
+                // Change the 'top' and 'left' CSS properties of flyLabel,
+                // triggering its CSS transition.
+                window.setTimeout(function(){
+                    console.log("flyLabel destinationElement.clientLeft", destinationElement.clientLeft);
+                    console.log("flyLabel destinationElement.clientTop", destinationElement.clientTop);
+                    flyLabel.style.left = destinationElement.getBoundingClientRect().left + 'px';
+                    flyLabel.style.top = destinationElement.getBoundingClientRect().top + 'px';
+                }, 1);
+
+                // Remove flyLabel after its transition has elapsed.
+                window.setTimeout(function(){
+                    flyLabel.parentElement.removeChild(flyLabel);
+                }, 1500);
+
+            });
 
```

Comment:
Unfortunately because you didn't edit the line I can't put the comment in the right place, but I would like to put this on [row 28](https://github.com/codefordc/housing-insights/pull/414/files#diff-cb09ec43944a93232c7ef43b427c1e8fR28). 

Bug I found: 
When clicking around with trying multiple filters, sometimes the filter does not place the proper pillbox. I couldn't quite figure out the pattern to reproduce the issue, but try: 
1) Slide the 'total units' slider
2) slide the 'subsidized units' slider
3) clear the 'total units' pillbox with the 'x'

Expected behavoir: total units pillbox removed
Actual behavior: 'total units' pillbox remains and 'subsidized units' dissappears. 

It looks like the clear button is actually clearing the right filter (see the console log from the State module - it publishes an empty proj_units_tot).

In a related issue, if you keep clicking around adding/removing pillboxes you can also sometimes end up with a situation where the code adds multiple copies of an active pillbox that is not the one you are currently clicking (e..g ACS Median Rent is added 5 times). I couldn't come up with a good click pattern to recreate this, but after enough playing with adding/removing I was able to make it happen every time. I think the pattern was re-adding ones that were above the last one after removing some in the middle/end. 

I think that the problem here is with the data binding that is occuring between the pillboxes and the activeFilterIds list. Since there is no key function, we are currently binding on the order of the pillboxes and the order of the `activeFilterIds` list. 

My guess is that adding a key function that binds the data and pillboxes based on the 'source' field will solve this problem: 
```
.selectAll('.clear-single',function (d) {
  return d['component']['source'];
})
```